### PR TITLE
Add an extra choice to available outcomes(determination)

### LINKED ISSUE #2533: Remove default from determination forms
## User story

As a user I would send out the wrong determination type because the default is either 'dismiss', 'accept', or 'need more information' depending on the workflow status. 

## Describe the solution you'd like in Hypha

I would like to select the determination type  I need (dismiss, accept, need more info) before sending out the determination to minimize human error.

Screen Shot 2021-09-03 at 3 29 36 PM

**Priority**
- Low priority (annoying, would be nice to not see)

**Affected roles**
- Applicants

**Ideal deadline**
December 2021

### PR DESCRIPTION
**Pull request by sandeepsajan0**
Fixes #2533 

Add description here.

### CODE REVIEW

File: hypha/apply/determinations/models.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ class Determination(DeterminationFormFieldsMixin, AccessFormData, models.Model):
         on_delete=models.PROTECT,
     )
 
-    outcome = models.IntegerField(verbose_name=_("Determination"), choices=DETERMINATION_CHOICES, default=1)
+    outcome = models.IntegerField(verbose_name=_("Determination"), choices=DETERMINATION_CHOICES)
```
Comment (sks444) [inline hypha/apply/determinations/models.py]: Let's not do any db level changes?

Looks like the issue only requires UI/forms level changes.

_changing default would require us to have a discussion on why it was set in the first place..and what could be the implications of changing it now_

File: hypha/apply/determinations/forms.py [hunk 2]
```diff
@@ -472,6 +475,12 @@ def clean(self):
 
         return cleaned_data
 
+    def clean_outcome(self):
+        if not self.cleaned_data['outcome']:
+            self.add_error('outcome', 'Outcome is a required field')
+            return self.cleaned_data
+        return int(self.cleaned_data['outcome'])
```
Comment (sks444) [inline hypha/apply/determinations/forms.py]: I guess this wouldn't be required if we do not remove the default from model field?

**Comment by sks444** [comment]
Looks good @sandeepsajan0.

Let's also do the same [for api ](https://github.com/HyphaApp/hypha/blob/main/hypha/apply/api/v1/determination/utils.py#L17) and make sure that react dashboard has the same behaviour as well.

**Comment by sandeepsajan0** [comment]
@sks444  I checked with one of the submission in my local, and it is working fine in react dashboard as well. Create button is disabled until we will not select a valid choice.
image

**Comment by sks444** [comment]
Thanks @sandeepsajan0!

**Comment by frjo** [comment]
@sandeepsajan0 This works perfectly!

I made the strings translatable and made the empty choice the first in the select drop down. I also added an empty choice to the proposal form dropdown.

Can you take a look and see if I implemented it in a sensible way, thanks!

**Comment by sandeepsajan0** [comment]
Thanks @frjo , Yeah It looks better now.

**Pull request merged**